doc/application/application.rst

```diff
@@ -675,15 +675,48 @@ following procedure:
 #. Rebuild the application normally following the steps specified
    in `Build an Application`_ above.
 
-
+.. _application_run:
 
 Run an Application
 ******************
 
-An application image can be run on real or emulated hardware. The kernel has
-built-in emulator support for QEMU. It allows you to run and test an application
-virtually, before (or in lieu of) loading and running it on actual target
-hardware.
+An application image can be run on real or emulated hardware.
+
+Most boards supported by Zephyr let you flash a compiled binary using
+the ``flash`` target to copy the binary to the board and run it.
+Follow these instructions to flash and run an application on real
+hardware:
+
+#. Open a terminal console and navigate to the application directory
+   :file:`~/app`.
+
+#. Enter the following command to build and run the application on
+   your board, which you usually must attach via USB.
+
+   .. code-block:: console
+
+       $ make [BOARD=<type> ...] flash
+
+   The Zephyr build system integrates with the board support files to
+   use hardware-specific tools to flash the Zephyr binary to your
+   hardware, then run it.
+
+   In cases where board support is incomplete, ``make flash`` may not
+   be supported. If you receive an error message about flash support
+   being unavailable when running this command, consult :ref:`your
+   board's documentation <boards>` for additional information on how
+   to flash your board.
+
+.. note:: When developing on Linux, it's common to need to install
+          board-specific udev rules to enable USB device access to
+          your board as a non-root user. If ``make flash`` fails,
+          consult your board's documentation to see if this is
+          necessary.
+
+The kernel has built-in emulator support for QEMU. It allows you to
```

Comment:
Is it the kernel or the SDK that has this QEMU support?  (I think the latter).

Reply:
Note that I haven't altered this text; this sentence appeared as-is before I touched this file.

The kernel does have QEMU support, in the form of the qemu_xxx boards:

```
boards/arm/qemu_cortex_m3  boards/riscv32/qemu_riscv32  boards/xtensa/qemu_xtensa
boards/nios2/qemu_nios2    boards/x86/qemu_x86
```

If this sentence needs changing, could fixing it be handled separately from this PR? That's unrelated to documenting `make flash`